### PR TITLE
Document image labels

### LINKED ISSUE #733: Add content to the "Add labels to the application image" page
## Summary

Add more content and examples (if applicable) to the [Add labels to the application image](https://github.com/buildpacks/docs/blob/main/content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md) page 

## What questions should this documentation answer?
- What is a label?
- Use cases
- Examples, if any

## Who is the target audience for this documentation?

- [ ] App Developer
- [x] Buildpack Author
- [ ] Operator
- [ ] Other:

### PR DESCRIPTION
**Pull request by hyounes4560**
- Document image labels
- Edits the exec.d page 

Closes #733

### CODE REVIEW

**Comment by hyounes4560** [comment]
@AidanDelaney content esp. examples might need an update.

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md [hunk 1]
```diff
@@ -5,6 +5,56 @@ weight=99
 
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+Labels are key-value pairs, stored as strings, that are attached to an image (i.e., arbitrary metadata). Labels are used to add helpful descriptions or attributes to an application image, which are meaningful to users.
 
-If you are familiar with this content and would like to make a contribution, please feel free to open a PR :)
+Labels are usually added at the time an image is created and can also be modified at a later time. Images can have multiple labels; however, each key must be unique.
```
Comment (AidanDelaney) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md:10]:
```suggestion
Labels are usually added at the time an image is created. Images can have multiple labels; however, each key must be unique.
```

In the buildpacks world, a buildpack writes labels to the `launch.toml` for the set of layers that it creates.  This is specified in https://github.com/buildpacks/spec/blob/main/buildpack.md#launchtoml-toml.  The contents of the various `launch.toml`s are assembled by `pack` (or `kpack` i.e. your _platform_) into the final output OCI Image.

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md [hunk 2]
```diff
@@ -5,6 +5,56 @@ weight=99
...
+* Including a description of the image and  additional metadata related to the image
+* Labels can also be used to organize images  
+
+## Implementation Steps
+
+* General syntax of `LABEL` instruction is as follows:
+
+    `LABEL <key-string>=<value-string>`
+
+    `LABEL version="1.0"`
```
Comment (AidanDelaney) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md:37]: Taking the perspective of a buildpack author, they would need to write 

```toml
[[labels]]
key-string = "value-string"
otherkey = "othervalue"
```

to `launch.toml` in their `<layers>/<layer>` directory.  If we use the running example from our Buildpack Author's Guide, this means writing to `"${CNB_LAYERS_DIR}"/node-js/launch.toml`.  However, it's worth noting that `launch.toml` also contains metadata about process and slices.

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md [hunk 3]
```diff
@@ -5,6 +5,56 @@ weight=99
...
+  "id": "<entrypoint buildpack ID>",
+  "version": "<entrypoint buildpack version>",
+  "stacks": [
+    {
+      "id": "<stack ID>",
+      "mixins": ["<mixin name>"]
+    }
+  ]
+}
+```
```
Comment (AidanDelaney) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md:60]: This example is important.  But I don't think it's the right example at this point.  If we stick with providing an exampl in shell, then the example could be a heredoc

```bash
cat << EOF > "${CNB_LAYERS_DIR}"/node-js/launch.toml
[[labels]]
key = "value"
EOF
```

An example in Go, would set the `Labels` field in a `libcnb.BuildResult` as returned by the `build` binary:

```go
func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
    result := libcnb.BuildResult{}

    result.Labels = append(result.Labels, libcnb.Label{Key: "key", Value: "value"})

    return result
}
```

The `libcnb` library will automatically take care of correctly writing `launch.toml` to the correct place.

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 4]
```diff
@@ -7,22 +7,24 @@ weight=99
...
+## Key Points
 
-    1. Location and Naming: Scripts are placed in the `<layer>/exec.d/` directory within a launch layer and must be executable. They can have any name.
+1. Location and Naming: Scripts are placed in the `<layer>/exec.d/` directory within a launch layer and must be executable. They can have any name.
 
-    2. Script Behavior:
+2. Script Behavior:
     * **Inputs**
-        * A third open file descriptor (in addition to stdout and stderr).  The third open file descriptor is inherited from the calling process.
+        *A third open file descriptor. File descriptors are integers used by a process to uniquely identify opened files; pre-opened file descriptors are usually 0 for stdin, 1 for stdout and 2 for stderr. The third open file descriptor is inherited from the calling process.
```
Comment (AidanDelaney) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:16]: I like this!

**Review by AidanDelaney — CHANGES_REQUESTED** [review]
There are some _very_ strong contributions here.  We need only clean up some of the technical language around how labels are implemented according to the buildpacks spec.

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md (hunk 3 above)
Comment (hyounes4560) [reply content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md:60]: @AidanDelaney should I remove the first example I added or add all 3? thanks

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md (hunk 2 above)
Comment (hyounes4560) [reply content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md:37]: not sure I get this? apologies

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md (hunk 1 above)
Comment (hyounes4560) [reply content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md:10]: updated

**Review by AidanDelaney — APPROVED** [review]
LGTM

**Comment by natalieparellano** [comment]
Just catching up here. Awesome work @hyounes4560!

**Pull request merged**